File: z-clients/javascript/src/models/msgQueueConfigureIn.ts
// this file is @generated

export interface MsgQueueConfigureIn {
    namespace?: string | null;
    retrySchedule?: number[];
    dlqTopic?: string | null;
}

export interface MsgQueueConfigureIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    retrySchedule?: number[];
    dlqTopic?: string | null;
}

export const MsgQueueConfigureInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgQueueConfigureIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            retrySchedule: object['retry_schedule'],
            dlqTopic: object['dlq_topic'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgQueueConfigureIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'retry_schedule': self.retrySchedule,
            'dlq_topic': self.dlqTopic,
        };
    }
}

File: z-clients/python/coyote/models/msg_queue_configure_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgQueueConfigureIn(BaseModel):
    namespace: t.Optional[str] = None

    retry_schedule: t.Optional[t.List[int]] = None

    dlq_topic: t.Optional[str] = None


class _MsgQueueConfigureIn(BaseModel):
    namespace: t.Optional[str] = None

    topic: str

    consumer_group: str

    retry_schedule: t.Optional[t.List[int]] = None

    dlq_topic: t.Optional[str] = None
